### PR TITLE
Feature local storage

### PR DESCRIPTION
**Pull request by michaelamponsah**
**In the feature implementation, the browser's local storage is used to persist user data 
entered in the contact me `section`.**

### CODE REVIEW

File: js/script.js [hunk 1]
```diff
@@ -89,4 +64,53 @@ document.addEventListener('click', (e) => {
...
+  }
+
+  const userData = {
+    userName: fullName.value,
+    userEmail: email.value,
+    userMessage: message.value,
+  };
+
+  storeDataInBrowser('userData', userData);
+});
```
Comment (Gopxfs) [inline js/script.js:107]: As requested in the project requirements:
image

Kindly notice that the user data should be stored after any changes to the form, not after submitting it 😉.

File: js/script.js [hunk 2]
```diff
@@ -89,4 +64,53 @@ document.addEventListener('click', (e) => {
...
+  if (email.validity.valueMissing) {
+    emailError.textContent = 'Email is required please';
+    email.setCustomValidity('');
+  } else if (email.validity.typeMismatch) {
+    emailError.textContent = 'Please provide a valid email';
+  } else if (email.validity.patternMismatch) {
+    emailError.textContent = 'Email should be in lowercase';
+  }
+  emailError.className = 'error error-active';
+}
```
Comment (Gopxfs) [inline js/script.js:79]: [OPTIONAL] Kindly notice that your email validation isn't working as expected. I tried to add an email containing a number, and received this error:
image

Then, after removing the numbers, the error still persisted and I had to reload the page:
image

Cheers and keep coding 😄.

**Review by Gopxfs — CHANGES_REQUESTED** [review]
# ♻️ STATUS: REQUIRED CHANGES ♻️

## Hi @michaelamponsah @Wolfmikerson 👋,

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights :rocket:
✔️ No linter errors.
✔️ Great code structure. Well done!
✔️ Data being stored after submitting the form.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions 🗒️  
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @Gopxfs in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: js/script.js [hunk 3]
```diff
@@ -89,4 +64,75 @@ document.addEventListener('click', (e) => {
...
+});
+
+const {
+  userName,
+  userEmail,
+  userMessage,
+} = getBrowserStoredData('userData');
+fullName.value = userName;
+email.value = userEmail;
+message.value = userMessage;
```
Comment (christianbiring1) [inline js/script.js:138]: [OPTIONAL]
- [ ] You may also consider fixing this error when the page load and the user has not yet put anything in the form inputs.

image

**Review by christianbiring1 — APPROVED** [review]
Hi @michaelamponsah , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
# STATUS: APPROVED :tada: 
nicee

## Highlights
- [ ] All linters are passing :heavy_check_mark: 
- [ ] Descriptive PR `summary` and `title`
- [ ] The project is working as expected :+1: 
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**